packages/forms/src/fields/MultiThumbRange.spec.js

```diff
@@ -0,0 +1,462 @@
+      const { getByTestId } = renderRtl(
+        <div style={{ width: 500 }}>
+          <MultiThumbRange minValue={15} maxValue={75} />
+        </div>
+      );
+
+      expect(getByTestId('track')).toHaveStyle('background-size: 60px; background-position: 25px;');
+    });
+  });
+
+  describe('Min Thumb', () => {
+    it('applies correct accessibility values', () => {
+      const { getAllByTestId } = render(<MultiThumbRange minValue={15} maxValue={75} />);
+      const thumb = getAllByTestId('thumb')[0];
+
+      expect(thumb).toHaveAttribute('role', 'slider');
+      expect(thumb).toHaveAttribute('tabIndex', '0');
+      expect(thumb).toHaveAttribute('aria-valuemin', '0');
+      expect(thumb).toHaveAttribute('aria-valuemax', '75');
+      expect(thumb).toHaveAttribute('aria-valuenow', '15');
+      expect(thumb).toHaveAttribute('aria-valuetext', '15');
+    });
+
+    it('removes thumb from tab order when disabled', () => {
+      const { getAllByTestId } = render(<MultiThumbRange disabled minValue={15} maxValue={75} />);
+      const thumb = getAllByTestId('thumb')[0];
+
+      expect(thumb).toHaveAttribute('tabIndex', '-1');
+    });
+
+    it('applies correct style', () => {
+      const { getAllByTestId } = render(<MultiThumbRange minValue={15} maxValue={75} />);
+      const thumb = getAllByTestId('thumb')[0];
+
+      expect(thumb).toHaveStyle('left: 15px');
+    });
+
+    it('applies correct style when in RTL mode', () => {
+      const { getAllByTestId } = renderRtl(<MultiThumbRange minValue={15} maxValue={75} />);
+      const thumb = getAllByTestId('thumb')[0];
+
+      expect(thumb).toHaveStyle('right: 15px');
+    });
+
+    describe('Key Handlers (LTR)', () => {
+      let onChangeSpy;
+
+      beforeEach(() => {
+        onChangeSpy = jest.fn();
+      });
+
+      it('decrements minValue on LEFT key', () => {
+        const { getAllByTestId } = render(
+          <MultiThumbRange minValue={15} maxValue={75} step={5} onChange={onChangeSpy} />
+        );
+
+        fireEvent.keyDown(getAllByTestId('thumb')[0], { keyCode: KEY_CODES.LEFT });
+        expect(onChangeSpy).toHaveBeenCalledWith({ minValue: 10, maxValue: 75 });
+      });
+
+      it('decrements minValue on DOWN key', () => {
+        const { getAllByTestId } = render(
+          <MultiThumbRange minValue={15} maxValue={75} step={5} onChange={onChangeSpy} />
+        );
+
+        fireEvent.keyDown(getAllByTestId('thumb')[0], { keyCode: KEY_CODES.DOWN });
+        expect(onChangeSpy).toHaveBeenCalledWith({ minValue: 10, maxValue: 75 });
+      });
+
+      it('increments minValue on RIGHT key', () => {
+        const { getAllByTestId } = render(
+          <MultiThumbRange minValue={15} maxValue={75} step={5} onChange={onChangeSpy} />
+        );
+
+        fireEvent.keyDown(getAllByTestId('thumb')[0], { keyCode: KEY_CODES.RIGHT });
+        expect(onChangeSpy).toHaveBeenCalledWith({ minValue: 20, maxValue: 75 });
+      });
```

Comment:
> priority querying

We use the role and text queries more often in `react-containers`. Since many of our consumers need to test via `data-test-id` attributes we use the same pattern internally to catch areas that we forget to spread props to their children.

Reply:
> render against changed output

This is how we (try) handle the testing in our more modern components. The tests within the `react-ranges` packages have gone through our `enzyme -> @testing-library/react` migration so the tests don't follow the best practices for integration testing. Since this is a hook-conversion I didn't want to mess with the tests as much as possible to avoid regressions.

Reply:
Thanks for the explanation @austin94!